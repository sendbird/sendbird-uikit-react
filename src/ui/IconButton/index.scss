@import '../../styles/variables';

.sendbird-iconbutton {
  color: var(--sendbird-iconbutton-color);
  display: inline-flex;
  justify-content: center;
  align-items: center;
  padding: 3px;
  background-color: transparent;
  cursor: pointer;
  user-select: none;
  border: 1px solid transparent;
  border-radius: 4px;
  outline: 0px;
  .sendbird-iconbutton__inner {
    height: 100%;
    svg {
      @include themed() {
        fill: t(on-bg-1);
      }
    }
  }

  &:hover {
    @include themed() {
      background-color: t(bg-1);
    }
  }

  &:active {
    @include themed() {
      background-color: t(bg-2);
    }
  }

  &:disabled {
    cursor: not-allowed !important;
    .sendbird-iconbutton__inner {
      svg, path {
        @include themed() {
          fill: t(on-bg-2);
        }
      }
    }
  }
}

.sendbird-iconbutton--pressed {
  @include themed() {
    background-color: t(primary--1-bg--7);
  }
}
