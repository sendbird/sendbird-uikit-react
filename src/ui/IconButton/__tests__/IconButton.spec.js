import React from 'react';
import { render } from '@testing-library/react';

import IconButton from "../index";

import DefaultIcon from '../../../svgs/icon-create.svg';

describe('ui/IconButton', () => {
  it('should do a snapshot test of the icon button DOM', function () {
    const { asFragment } = render(
      <IconButton>
        <DefaultIcon />
      </IconButton>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
