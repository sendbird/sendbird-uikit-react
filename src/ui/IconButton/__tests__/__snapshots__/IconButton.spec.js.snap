// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/IconButton should do a snapshot test of the icon button DOM 1`] = `
<DocumentFragment>
  <button
    class=" sendbird-iconbutton "
    style="height: 56px; width: 56px;"
    type="button"
  >
    <span
      class="sendbird-iconbutton__inner"
    >
      <test-file-stub />
    </span>
  </button>
</DocumentFragment>
`;
